src/main/java/com/gmail/goosius/siegewar/listeners/SiegeWarTownEventListener.java

```diff
@@ -339,8 +339,12 @@ public void onTownStatusScreen(TownStatusScreenEvent event) {
 	        if (SiegeController.hasSiege(town)) {
 				Siege siege = SiegeController.getSiege(town);
 				SiegeStatus siegeStatus= siege.getStatus();
-				String time = TimeMgmt.getFormattedTimeValue(TownMetaDataController.getSiegeImmunityEndTime(town)- System.currentTimeMillis());
-
+				String time;
+				if (TownMetaDataController.getSiegeImmunityEndTime(town) != -1l) {
+					time = TimeMgmt.getFormattedTimeValue(TownMetaDataController.getSiegeImmunityEndTime(town)- System.currentTimeMillis());
+				} else {
+					time = "Permanent";

```

Comment:
:construction: 
- You need to get the translation here, rather than hardcode the word "permanent"

Reply:
:construction: 
This still needs to be done.